### PR TITLE
support to_static for SpectralNorm

### PR DESCRIPTION
**Pull request by wangna11BD**
### PR types
Bug fixes

### PR changes
APIs

### Describe
support to_static for SpectralNorm

### CODE REVIEW

File: python/paddle/nn/utils/spectral_norm_hook.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 def normal_(x, mean=0.0, std=1.0):
     temp_value = paddle.normal(mean, std, shape=x.shape)
-    x.set_value(temp_value)
+    x = paddle.assign(temp_value)
```
Comment (LielinJiang) [inline python/paddle/nn/utils/spectral_norm_hook.py]: assign要加一下x，否则这里的x是一个新的变量

File: python/paddle/nn/utils/spectral_norm_hook.py [hunk 2]
```diff
@@ -61,7 +61,7 @@ def compute_weight(self, layer, do_power_iteration):
         if do_power_iteration:
             with paddle.no_grad():
                 for _ in range(self.n_power_iterations):
-                    v.set_value(
+                    v = paddle.assign(
```
Comment (LielinJiang) [inline python/paddle/nn/utils/spectral_norm_hook.py]: same above

File: python/paddle/nn/utils/spectral_norm_hook.py (hunk 1 above)
Comment (wangna11BD) [reply python/paddle/nn/utils/spectral_norm_hook.py]: Done

File: python/paddle/nn/utils/spectral_norm_hook.py (hunk 2 above)
Comment (wangna11BD) [reply python/paddle/nn/utils/spectral_norm_hook.py]: Done

**Review by LielinJiang — DISMISSED** [review]

**Review by LielinJiang — APPROVED** [review]

**Pull request merged**